src/Tizen.BatteryMonitor/Tizen.BatteryMonitor/BatteryMonitorEnumerations.cs

```diff
@@ -0,0 +1,60 @@
+﻿/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+using System;
+
+namespace Tizen.BatteryMonitor
+{
+    /// <summary>
+    /// Enumeration for battery consuming features.
+    /// </summary>
+    /// <since_tizen> 7 </since_tizen>
+    public enum ResourceType {
+        /// <summary>
+        /// Bluetooth Low Energy.
+        /// </summary>
+        Ble = 0,
+
+        /// <summary>
+        /// Wi-Fi.
+        /// </summary>
+        Wifi = 1,
+
+        /// <summary>
+        /// CPU.
+        /// </summary>
+        Cpu = 2,
+
+        /// <summary>
+        /// Display.
+        /// </summary>
+        Display = 3,
+
+        /// <summary>
+        /// Device Network.
+        /// </summary>
+        DeviceNetwork = 4,
+
+        /// <summary>
+        /// GPS Sensor.
+        /// </summary>
+        Gps = 5,
+
+        /// <summary>
+        /// Use for iterating over enums only.
+        /// </summary>
+        Max
```

Comment:
In C #, enum can traversed without max length.
You can use Enum.GetValues and also don't need length because you have foreach.
Also, in C, enum is just the same as int,
The interop function does not need to have an enum with exactly the same value If you're not really using MAX as an argument.

Wouldn't it be nice to have no max?

Reply:
yes. will remove Max. Thanks.